{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import annotations\n",
    "import pysam\n",
    "import os\n",
    "import pandas\n",
    "import typing\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/ceph01/homedirs/brand/Projects/extreme-memorizers/playbooks'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "VCF_FILE = \"../output/102-08068/102-08068.hard-filtered.annotated.vcf.gz\"\n",
    "MIN_CADD = 24\n",
    "MAX_GNOMAD_AF = 0\n",
    "IMPACT = [ 'HIGH', 'MODERATE' ]\n",
    "LOF_METRICS = \"../external_scripts/files/gnomad.v2.1.1.lof_metrics.by_gene.txt\"\n",
    "MIN_PLI = 0.9\n",
    "MIN_MIS_Z = 3.09\n",
    "MAX_OE_LOF = 0.35"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "apply_fn = typing.Callable[[pysam.libcbcf.VariantRecord], pysam.libcbcf.VariantRecord]\n",
    "filter_fn = typing.Callable[[pysam.libcbcf.VariantRecord], bool]\n",
    "record_type = pysam.libcbcf.VariantRecord\n",
    "\n",
    "\n",
    "record_attributes = {\n",
    "    'alts': 'list',\n",
    "    'chrom': 'fixed',\n",
    "    'filter': 'list',\n",
    "    'format': 'list',\n",
    "    'id': 'fixed',\n",
    "    'info': 'dict',\n",
    "    'pos': 'fixed',\n",
    "    'qual': 'fixed',\n",
    "    'ref': 'fixed',\n",
    "    'samples': 'dict',\n",
    "}\n",
    "\n",
    "\n",
    "class FilterIteratorException(Exception):\n",
    "    pass\n",
    "\n",
    "\n",
    "class RecordAttributeError(Exception):\n",
    "    pass\n",
    "\n",
    "\n",
    "class FilterVariants():\n",
    "    class _TransformFn():\n",
    "        def __init__(self, fn: typing.Union[apply_fn, filter_fn], is_filter: bool = True):\n",
    "            self._fn = fn\n",
    "            self.is_filter = is_filter\n",
    "        \n",
    "        def __call__(self, record: record_type):\n",
    "            return self._fn(record)\n",
    "        \n",
    "    def __init__(self, vcf_file: str):\n",
    "        self._vcf = pysam.VariantFile(vcf_file)\n",
    "        self._transforms = []\n",
    "        self._is_iterating = False\n",
    "        self._filtered_records = 0\n",
    "        self._annotation_header = None\n",
    "        \n",
    "    def apply(self, fn: apply_fn) -> FilterVariants:\n",
    "        if self._is_iterating:\n",
    "            raise FilterIteratorException(\"Changing transformations during iteration.\")\n",
    "        self._transforms.append(FilterVariants._TransformFn(fn, is_filter=False))\n",
    "        return self\n",
    "        \n",
    "    def filter(self, fn: fiter_fn) -> FilterVariants:\n",
    "        if self._is_iterating:\n",
    "            raise FilterIteratorException(\"Changing transformations during iteration.\")\n",
    "        self._transforms.append(FilterVariants._TransformFn(fn, is_filter=True))\n",
    "        return self\n",
    "        \n",
    "    def _apply_transforms(self, record: record_type) -> typing.Tuple[bool, record_type]:\n",
    "        is_filtered = False\n",
    "        for fn in self._transforms:\n",
    "            if is_filtered:\n",
    "                break\n",
    "            if fn.is_filter:\n",
    "                is_filtered = not fn(record)\n",
    "            else:\n",
    "                record = fn(record)\n",
    "        return (is_filtered, record)\n",
    "        \n",
    "    @property\n",
    "    def records(self):\n",
    "        for record in self._vcf.fetch():\n",
    "            self._is_iterating = True\n",
    "            is_filtered, record = self._apply_transforms(record)\n",
    "            if not is_filtered:\n",
    "                yield record\n",
    "            else:\n",
    "                self._filtered_records += 1\n",
    "        self._is_iterating = False\n",
    "    \n",
    "    @property\n",
    "    def annotation_header(self) -> typing.List[str]:\n",
    "        if self._annotation_header is None:\n",
    "            self._annotation_header = list(filter(lambda x: x.get('ID') == 'CSQ', self._vcf.header.records))[0].get('Description').split(':')[1].strip(\" \\\"\").split('|')\n",
    "        return self._annotation_header\n",
    "    \n",
    "    @classmethod\n",
    "    def get_annotations(cls, record: record_type, header: typing.List[str], info_field='CSQ') -> typing.List[typing.Mapping[str, str]]:\n",
    "        def _parse_val(val: str) -> typing.Union[str, float, int]:\n",
    "            if re.match(r\"^[+-]?\\d+$\", val):\n",
    "                return int(val)\n",
    "            elif re.match(r'^[+-]?(\\d+(\\.\\d*)?|\\.\\d+)([eE][+-]?\\d+)?$', val):\n",
    "                return float(val)\n",
    "            elif val == '':\n",
    "                return None\n",
    "            else:\n",
    "                return val\n",
    "        return [{\n",
    "            header[i]: _parse_val(k) for i, k in enumerate(csq_field.split('|'))\n",
    "        } for csq_field in record.info[info_field] ]\n",
    "    \n",
    "    def _sample_to_row(self,\n",
    "        record: record_type,\n",
    "        attributes: typing.List[str] = None,\n",
    "    ) -> typing.Tuple[typing.List[typing.List[str]], typing.List[str]]:\n",
    "        rows = []\n",
    "        if attributes is None:\n",
    "            s = list(record.samples)[0]\n",
    "            s_attrs = set(list(record.samples[s]))\n",
    "        else:\n",
    "            attrs = set(filter(lambda s: s.startswith('samples'), attributes))\n",
    "            s_attrs = set(map(lambda s: s.split('.')[1], attrs))\n",
    "        header = None\n",
    "        for s in list(record.samples):\n",
    "            attrs_to_row = set(list(record.samples[s])) & s_attrs\n",
    "            header = list(sorted(attrs_to_row))\n",
    "            rows.append([\n",
    "                s,\n",
    "                *[ record.samples[s][a] for a in sorted(attrs_to_row) ]\n",
    "            ])\n",
    "        return rows, ['sample_id', *[ f'sample.{a}' for a in header ]]\n",
    "            \n",
    "    def _attrs_to_row(self,\n",
    "        record: record_type,\n",
    "        attributes: typing.List[str] = None,\n",
    "        excl_attr: typing.List[str] = [ 'samples' ],\n",
    "        anno_field: str = 'CSQ',\n",
    "        split_anno: bool = True,\n",
    "    ) -> typing.Tuple[typing.List[typing.List[str]], typing.List[str]]:\n",
    "        \n",
    "        def _get_single_row(record, attrs, anno_field=anno_field) -> typing.List[str]:\n",
    "            row = []\n",
    "            for a in attrs:\n",
    "                a_key = a.split('.')[0]\n",
    "                if record_attributes[a_key] == 'fixed':\n",
    "                    row.append(getattr(record, a_key))\n",
    "                elif record_attributes[a_key] == 'list':\n",
    "                    row.append(','.join(list(getattr(record, a_key))))\n",
    "                elif record_attributes[a_key] == 'dict':\n",
    "                    attr_value = getattr(record, a_key)\n",
    "                    if len(a.split('.')) == 1:\n",
    "                        # Get all\n",
    "                        for key in list(attr_value):\n",
    "                            if a_key == 'info' and key == anno_field:\n",
    "                                continue\n",
    "                            else:\n",
    "                                row.append(attr_value[key])\n",
    "                    elif a.split('.')[1] in list(attr_value):\n",
    "                        row.append(attr_value[a.split('.')[1]])\n",
    "                    else:\n",
    "                        raise RecordAttributeError(f\"{a} could not be found on record.\")\n",
    "            return row\n",
    "\n",
    "        if attributes is not None:\n",
    "            attrs = set(filter(lambda s: s.split('.')[0] in record_attributes.keys())) - set(excl_attr)\n",
    "        else:\n",
    "            attrs = set(record_attributes.keys()) - set(excl_attr)\n",
    "\n",
    "        has_anno = any(a.startswith(f'info.{anno_field}') for a in attrs)\n",
    "        if has_anno or attributes is None:\n",
    "            anno = FilterVariants.get_annotations(record, self.annotation_header)\n",
    "        else:\n",
    "            anno = None\n",
    "            \n",
    "        if anno is None:\n",
    "            return [ _get_single_row(record, attrs) ]\n",
    "        else:\n",
    "            if len(set(list(record.info)) - set([ anno_field ])) == 0:\n",
    "                attrs -= set([ 'info' ])\n",
    "            return [\n",
    "                [ *a.values(), *_get_single_row(record, attrs) ] for a in anno \n",
    "            ], [ *[ f\"info.{a}\" for a in anno[0].keys() ] , *attrs ]\n",
    "\n",
    "    \n",
    "    def _record_to_row(self,\n",
    "        record: record_type,\n",
    "        attributes: typing.List[str] = None,\n",
    "        info_field: str = 'CSQ',\n",
    "        split_info: bool = True\n",
    "    ) -> typing.Tuple[typing.List[typing.List[str]], typing.List[str]]:\n",
    "        rows = []\n",
    "        sample_rows = self._sample_to_row(record, attributes)\n",
    "        row_attrs = self._attrs_to_row(record, attributes, anno_field=info_field, split_anno=split_info)\n",
    "        var_id = f\"{record.contig}_{record.start}_{record.ref}_{record.alts[0]}\"\n",
    "        header = [ 'var_id', *row_attrs[1], *sample_rows[1] ]\n",
    "        for s in sample_rows[0]:\n",
    "            for a in row_attrs[0]:\n",
    "                rows.append([\n",
    "                    var_id,\n",
    "                    *a,\n",
    "                    *s,\n",
    "                ])\n",
    "        return rows, header \n",
    "    \n",
    "    def to_pandas(self, attributes: typing.List[str] = None) -> pandas.DataFrame:\n",
    "        rows = []\n",
    "        header = []\n",
    "        for record in self.records:\n",
    "            data, header = self._record_to_row(record, attributes)\n",
    "            rows = [ *rows, *data ]\n",
    "        return pandas.DataFrame(rows, columns=header)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 133969487 [None, None, None] [21.8, 21.8, 21.8] ['HIGH', 'MODIFIER', 'MODIFIER'] ['frameshift_variant', '5_prime_UTR_variant', 'upstream_gene_variant']\n",
      "5 60628574 [0] [12.31] ['MODERATE'] ['inframe_deletion']\n"
     ]
    }
   ],
   "source": [
    "filter_obj = FilterVariants(VCF_FILE)\n",
    "lof = pandas.read_csv(LOF_METRICS, sep=\"\\t\").set_index(\"gene\", drop=False)\n",
    "\n",
    "\n",
    "def filter_cadd(\n",
    "    record: record_type,\n",
    "    header: typing.List[str] = filter_obj.annotation_header,\n",
    "    min_cadd: int = MIN_CADD\n",
    ") -> bool:\n",
    "    anno = FilterVariants.get_annotations(record, header)\n",
    "    return any('CADD_PHRED' in a.keys() and a['CADD_PHRED'] is not None and a['CADD_PHRED'] >= min_cadd for a in anno)\n",
    "\n",
    "def filter_gnomad(\n",
    "    record: record_type,\n",
    "    header: typing.List[str] = filter_obj.annotation_header,\n",
    "    max_gnomad_af: float = MAX_GNOMAD_AF\n",
    ") -> bool:\n",
    "    anno = FilterVariants.get_annotations(record, header)\n",
    "    return any('gnomAD_AF' in a and (a['gnomAD_AF'] is None or a['gnomAD_AF'] <= max_gnomad_af) for a in anno)\n",
    "\n",
    "def filter_impact(\n",
    "    record: record_type,\n",
    "    header: typing.List[str] = filter_obj.annotation_header,\n",
    "    impact: typing.List[str] = IMPACT,\n",
    ") -> bool:\n",
    "    anno = FilterVariants.get_annotations(record, header)\n",
    "    return any('IMPACT' in a and a['IMPACT'] is not None and a['IMPACT'] in impact for a in anno)\n",
    "\n",
    "def filter_constraint(\n",
    "    record: record_type,\n",
    "    header: typing.List[str] = filter_obj.annotation_header,\n",
    "    lof_metrics: pandas.DataFrame = lof,\n",
    "    min_pli: float = MIN_PLI,\n",
    "    min_mis_z: float = MIN_MIS_Z,\n",
    "    max_oe_lof: float = MAX_OE_LOF,\n",
    ") -> bool:\n",
    "    anno = FilterVariants.get_annotations(record, header)\n",
    "    gene = anno[0]['SYMBOL']\n",
    "    if gene in lof_metrics.index:\n",
    "        return any(\n",
    "                (a['Consequence'] == 'missense_variant' and lof_metrics.loc[gene].mis_z > min_mis_z)\n",
    "                or (lof_metrics.loc[gene].pLI >= min_pli and lof_metrics.loc[gene].oe_lof_upper < max_oe_lof)\n",
    "            for a in anno)\n",
    "    else:\n",
    "        return False\n",
    "\n",
    "filter_obj = filter_obj.filter(filter_impact).filter(filter_constraint).filter(filter_gnomad) #.filter(filter_cadd)\n",
    "\n",
    "for record in filter_obj.records:\n",
    "    anno = FilterVariants.get_annotations(record, filter_obj.annotation_header)\n",
    "    print(record.chrom, record.pos, [ a['gnomAD_AF'] for a in anno ], [ a['CADD_PHRED'] for a in anno ], [ a['IMPACT'] for a in anno ], [ a['Consequence'] for a in anno ])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = filter_obj.to_pandas()\n",
    "df = df[df['info.IMPACT'].apply(lambda s: s in IMPACT)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"filtered_vars.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.82026"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lof.loc['NGF']['pLI']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'Allele': '-',\n",
       "  'Consequence': 'intron_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000299421',\n",
       "  'BIOTYPE': 'protein_coding',\n",
       "  'EXON': None,\n",
       "  'INTRON': '5/12',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'downstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'TAF10',\n",
       "  'Gene': 'ENSG00000166337',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000299424',\n",
       "  'BIOTYPE': 'protein_coding',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 2044,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 11543,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'downstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'TPP1',\n",
       "  'Gene': 'ENSG00000166340',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000299427',\n",
       "  'BIOTYPE': 'protein_coding',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 3971,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 2073,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000396751',\n",
       "  'BIOTYPE': 'protein_coding',\n",
       "  'EXON': None,\n",
       "  'INTRON': '4/11',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000420936',\n",
       "  'BIOTYPE': 'protein_coding',\n",
       "  'EXON': None,\n",
       "  'INTRON': '5/12',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'downstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000524735',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 381,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant&NMD_transcript_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000526114',\n",
       "  'BIOTYPE': 'nonsense_mediated_decay',\n",
       "  'EXON': None,\n",
       "  'INTRON': '6/7',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant&non_coding_transcript_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000526318',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': '3/3',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'upstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000526711',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 143,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'non_coding_transcript_exon_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'TAF10',\n",
       "  'Gene': 'ENSG00000166337',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000526743',\n",
       "  'BIOTYPE': 'retained_intron',\n",
       "  'EXON': '3/3',\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': 2890,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 11543,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant&NMD_transcript_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000527121',\n",
       "  'BIOTYPE': 'nonsense_mediated_decay',\n",
       "  'EXON': None,\n",
       "  'INTRON': '6/8',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'upstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'RP11-732A19.8',\n",
       "  'Gene': 'ENSG00000254400',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000527191',\n",
       "  'BIOTYPE': 'antisense',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 4379,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'Clone_based_vega_gene',\n",
       "  'HGNC_ID': None,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'downstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'TAF10',\n",
       "  'Gene': 'ENSG00000166337',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000527248',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 2047,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 11543,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'frameshift_variant&NMD_transcript_variant',\n",
       "  'IMPACT': 'HIGH',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000527327',\n",
       "  'BIOTYPE': 'nonsense_mediated_decay',\n",
       "  'EXON': '5/12',\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': 582,\n",
       "  'CDS_position': 457,\n",
       "  'Protein_position': 153,\n",
       "  'Amino_acids': 'P/X',\n",
       "  'Codons': 'Ccc/cc',\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'frameshift_variant&NMD_transcript_variant',\n",
       "  'IMPACT': 'HIGH',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000527394',\n",
       "  'BIOTYPE': 'nonsense_mediated_decay',\n",
       "  'EXON': '5/11',\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': 598,\n",
       "  'CDS_position': 457,\n",
       "  'Protein_position': 153,\n",
       "  'Amino_acids': 'P/X',\n",
       "  'Codons': 'Ccc/cc',\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant&non_coding_transcript_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'RP11-732A19.2',\n",
       "  'Gene': 'ENSG00000254641',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000527398',\n",
       "  'BIOTYPE': 'sense_overlapping',\n",
       "  'EXON': None,\n",
       "  'INTRON': '2/2',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'Clone_based_vega_gene',\n",
       "  'HGNC_ID': None,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant&non_coding_transcript_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000528784',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': '2/7',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000528995',\n",
       "  'BIOTYPE': 'protein_coding',\n",
       "  'EXON': None,\n",
       "  'INTRON': '4/10',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'upstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000530016',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 81,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'downstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'TAF10',\n",
       "  'Gene': 'ENSG00000166337',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000531760',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 2390,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 11543,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant&NMD_transcript_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000532063',\n",
       "  'BIOTYPE': 'nonsense_mediated_decay',\n",
       "  'EXON': None,\n",
       "  'INTRON': '4/11',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'downstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'TAF10',\n",
       "  'Gene': 'ENSG00000166337',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000532344',\n",
       "  'BIOTYPE': 'retained_intron',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 2603,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 11543,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'downstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'TPP1',\n",
       "  'Gene': 'ENSG00000166340',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000533371',\n",
       "  'BIOTYPE': 'protein_coding',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 3974,\n",
       "  'STRAND': -1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 2073,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'downstream_gene_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000534565',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': None,\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': 4073,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant&non_coding_transcript_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000534706',\n",
       "  'BIOTYPE': 'processed_transcript',\n",
       "  'EXON': None,\n",
       "  'INTRON': '1/3',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0},\n",
       " {'Allele': '-',\n",
       "  'Consequence': 'intron_variant',\n",
       "  'IMPACT': 'MODIFIER',\n",
       "  'SYMBOL': 'ILK',\n",
       "  'Gene': 'ENSG00000166333',\n",
       "  'Feature_type': 'Transcript',\n",
       "  'Feature': 'ENST00000537806',\n",
       "  'BIOTYPE': 'protein_coding',\n",
       "  'EXON': None,\n",
       "  'INTRON': '4/11',\n",
       "  'HGVSc': None,\n",
       "  'HGVSp': None,\n",
       "  'cDNA_position': None,\n",
       "  'CDS_position': None,\n",
       "  'Protein_position': None,\n",
       "  'Amino_acids': None,\n",
       "  'Codons': None,\n",
       "  'Existing_variation': 'COSV54990603',\n",
       "  'DISTANCE': None,\n",
       "  'STRAND': 1,\n",
       "  'FLAGS': None,\n",
       "  'SYMBOL_SOURCE': 'HGNC',\n",
       "  'HGNC_ID': 6040,\n",
       "  'SOURCE': None,\n",
       "  'SIFT': None,\n",
       "  'PolyPhen': None,\n",
       "  'gnomAD_AF': None,\n",
       "  'gnomAD_AFR_AF': None,\n",
       "  'gnomAD_AMR_AF': None,\n",
       "  'gnomAD_ASJ_AF': None,\n",
       "  'gnomAD_EAS_AF': None,\n",
       "  'gnomAD_FIN_AF': None,\n",
       "  'gnomAD_NFE_AF': None,\n",
       "  'gnomAD_OTH_AF': None,\n",
       "  'gnomAD_SAS_AF': None,\n",
       "  'CLIN_SIG': None,\n",
       "  'SOMATIC': 1,\n",
       "  'PHENO': 1,\n",
       "  'CADD_PHRED': 1.914,\n",
       "  'CADD_RAW': -0.052206,\n",
       "  'gnomADg': 'rs11314683',\n",
       "  'gnomADg_AF_AFR': 1.0,\n",
       "  'gnomADg_AF_AMR': 1.0,\n",
       "  'gnomADg_AF_ASJ': 1.0,\n",
       "  'gnomADg_AF_EAS': 1.0,\n",
       "  'gnomADg_AF_FIN': 1.0,\n",
       "  'gnomADg_AF_NFE': 1.0,\n",
       "  'gnomADg_AF_OTH': 1.0}]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pysam",
   "language": "python",
   "name": "pysam"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
